### ISSUE TITLE
Add `minCommunitySize` to non-LPA community detection algorithms

### ISSUE THREAD

**Issue by FlorentinD** [neo4j/graph-data-science#243]
**Is your feature request related to a problem? Please describe.**

For Label Propagation and Louvain, we added the `minCommunitySize` to its write mode ([example](https://neo4j.com/docs/graph-data-science/current/algorithms/label-propagation/#algorithms-label-propagation-syntax)).

**Describe the solution you would like**

Additionally the `minCommunitySize` parameter could be exposed to:

- [ ] Triangle Count 
- [x] K1-coloring
- [x] ModularityOptimization
- [x] Approximate Maximum k-cut
- [x] Leiden
- [x] WCC.stream (as `minComponentSize`)
- [x] Louvain.stream
- [ ] LabelPropagation.stream

**Additional context**

The logic for the `minComponentSize|minCommunitySize` is contained inside the `CommunityProcCompanion`.

If you want to work on this issue please drop a comment :)

**Comment by airtyon** [comment]
hey @FlorentinD, I would like to give this one a try. I will ask here if I have any questions.

**Comment by airtyon** [comment]
hey @FlorentinD, will the new attributes be added to the stream & write mode of the listed algorithms(except the last 2 items) or just to the write mode? if it's the latest, I noticed that Approximate Maximum k-cut, doesn't a write mode. 
thank you.

**Comment by FlorentinD** [comment]
Hey @airtyon ,
I would say we can also add it to the stream modes.
If you like, you could also just do the write modes and we can adjust the checklist to reflect what is left after your PR :)

Thanks for picking this up!

**Comment by airtyon** [comment]
hey @FlorentinD,
I completed the write mode for the algorithms listed above, I have some doubts regarding Triangle Count & K1-coloring.

the reason I am commenting is that after adding minCommunitySize to Louvain in stream mode and afterwards trying to extends the tests and altering the value of minCommunitySize; I didn't see any change in the final output.

it is obvious that I have a knowledge gap somewhere but I can't pinpoint it.

I hope what I wrote makes sense.
